app/src/screen.c

```diff
@@ -300,8 +341,7 @@ screen_switch_fullscreen(struct screen *screen) {
     screen->fullscreen = !screen->fullscreen;
     if (!screen->fullscreen) {
         // fullscreen disabled, restore expected windowed window size
-        SDL_SetWindowSize(screen->window, screen->windowed_window_size.width,
-                          screen->windowed_window_size.height);
+        set_window_size(screen, screen->windowed_window_size);
```

Comment:
Just for info, why is it necessary to check (and possibility recreate the renderer) when we switch fullscreen? (same question for `resize_to_fit()` and `resize_to_pixel_perfect()`

Reply:
because the bug reproduces whether the window resize is caused by the user (by dragging the corner), but it also reproduces when it is resized by the system through a SDL_SetWindowSize call - and it is not picked up by the event in the event loop either.  Regardless of the rest of the patch I think this small refactor is good on its own, it puts all the calls to SDL_SetWindowSize in a single code path.

Reply:
> but it also reproduces when it is resized by the system through a SDL_SetWindowSize call - and it is not picked up by the event in the event loop either.

I don't understand well.

The problem does not occur only when moving the window from one monitor to another (which generates a `SDL_WINDOWEVENT_SIZE_CHANGED` event)?
If the user calls `SDL_SetWindowSize`, then the window should not be moved to another monitor?

Reply:
> Regardless of the rest of the patch I think this small refactor is good on its own, it puts all the calls to SDL_SetWindowSize in a single code path.

The purpose of `set_window_size()` is to be able to change the windowed size even when fullscreen is enabled.

We don't want to resize the windowed size by a shortcut while in fullscreen (it's confusing).